Add support for Portable games

ShrineFox opened the pull request:
- Added P3P, P4D, P4G to games list
- Added CPK extraction capabilities
- Added optional CPK compression

ShrineFox committed:
Add support for Portable games

- Added P3P, P4D, P4G to games list
- Added CPK extraction capabilities
- Added optional CPK compression

tge-was-taken commented on Source/ModCompendiumLibrary/ModSystem/Builders/IModBuilder.cs:
```diff
@@ -9,7 +9,8 @@ public interface IModBuilder
+        /// <param name="compression">Optional. If compression == "True", the mod builder will compress the resulting cpk.</param>
         /// <returns></returns>
-        VirtualFileSystemEntry Build( VirtualDirectory root, string hostOutputPath = null );
+        VirtualFileSystemEntry Build( VirtualDirectory root, string hostOutputPath = null, string compression = null );
```
Compression should be a bool. convert it to a string with .ToString()
Also change the name to `useCompression`. 
I like booleans to be prefixed with a verb as they represent state and it reads a bit more naturally.

tge-was-taken commented on Source/ModCompendiumLibrary/ModSystem/Builders/Persona5ModCpkModBuilder.cs:
```diff
@@ -9,7 +9,7 @@ namespace ModCompendiumLibrary.ModSystem.Builders
     {
         /// <inheritdoc />
-        public VirtualFileSystemEntry Build( VirtualDirectory root, string hostOutputPath = null )
+        public VirtualFileSystemEntry Build( VirtualDirectory root, string hostOutputPath = null, string csvPath = null )
```
Last parameter should be compression to match the interface declaration

tge-was-taken commented on Source/ModCompendiumLibrary/ModSystem/Builders/Persona5ModCpkModBuilder.cs:
```diff
@@ -69,7 +69,7 @@ public VirtualFileSystemEntry Build( VirtualDirectory root, string hostOutputPat
             var cpkModCompiler = new CpkModBuilder();
             var cpkFilePath = hostOutputPath != null ? Path.Combine( hostOutputPath, "mod.cpk" ) : null;
-            var cpkFile = cpkModCompiler.Build( modFilesDirectory, cpkFilePath );
+            var cpkFile = cpkModCompiler.Build( modFilesDirectory, cpkFilePath, string.Empty);
```
You're passing string.Empty instead of the compression parameter, why can't mod.cpks be compressed too?

tge-was-taken commented on Source/ModCompendiumLibrary/ModSystem/Builders/PersonaPortable/CsvMaker.cs:
```diff
@@ -0,0 +1,67 @@
+                Log.Config.Trace($"Loading CSV file: {baseCsv}");
+                try
+                {
+
```
?

tge-was-taken commented on Source/ModCompendiumLibrary/ModSystem/Builders/PersonaPortable/CsvMaker.cs:
```diff
@@ -0,0 +1,67 @@
+                        break;
+                    }
+                }
+                if (matchFound == false)
```
matchFound is a bool, there is no need to compare it to another boolean.
Should be `if (!matchFound)` where the `!` is the logical negation operator

tge-was-taken commented on Source/ModCompendiumLibrary/ModSystem/Builders/PersonaPortableModBuilder.cs:
```diff
@@ -0,0 +1,162 @@
+            // Get files from CPK
+            VirtualDirectory cpkRootDirectory;
+
+            Log.Builder.Trace($"RootOrCpkPath = {config.CpkRootOrPath}");
```
minor nit, but there's a discrepancy between the log text and the actual name of the variable.
you could solve this by using `nameof`

tge-was-taken commented on Source/ModCompendiumLibrary/ModSystem/Builders/PersonaPortableModBuilder.cs:
```diff
@@ -0,0 +1,162 @@
+                    throw new InvalidConfigException($"CPK root path references a CPK file that does not exist: {config.CpkRootOrPath}.");
+                }
+
+                string[] args = new string[] { "-x", "-i", config.CpkRootOrPath, "-d", cpkRootDirectoryPath };
```
little thing for you to know, you can omit `new string[]` and specify `new[]` instead of each of the elements in the array is of the same known type (eg string)

tge-was-taken commented on Source/ModCompendiumLibrary/ModSystem/Builders/PersonaPortableModBuilder.cs:
```diff
@@ -0,0 +1,162 @@
+            //If compression is enabled, pass game name to cpk builder
+            if (config.Compression == "True")
+            {
+                compression = Game.ToString();
```
not sure what the correlation between compression and game is

tge-was-taken reviewed (changes requested):
Looks good for the most part, but some things seem a bit questionable. I would've personally chosen a different CRI CPK library to use for extraction, but that can be left for later.

ShrineFox committed:
mod.cpk compression for P5

- useCompression is now a bool
- separated useCompression from gameName
- Compression checkbox is available in P5's config now

ShrineFox commented:
All requested changes have been addressed, pushed a new commit to the branch.

tge-was-taken commented on Source/ModCompendiumLibrary/ModSystem/Builders/PersonaPortableModBuilder.cs:
```diff
@@ -42,7 +44,7 @@ public VirtualFileSystemEntry Build(VirtualDirectory root, string hostOutputPath
             VirtualDirectory cpkRootDirectory;
 
-            Log.Builder.Trace($"RootOrCpkPath = {config.CpkRootOrPath}");
+            Log.Builder.Trace($"RootOrCpkPath = {nameof(config.CpkRootOrPath)}");
```
this will print "RootOrCpkPath = CpkRootOrPath". Im guessing you mixed up what I said.

ShrineFox replied:
Yup, my bad. Should be good now 👍

ShrineFox committed:
Missed the variable name mismatch

there we heckin go
